src/main/java/com/weather/discordweather/gateway/WeatherForecastGateway.java

```diff
@@ -54,4 +72,19 @@ public String executeWebhook(String forecast) {
         forecast
     );
   }
+
+  private Optional<Coordinate> extractCoordinates(GeocodeResponse response) {
+    if (response.results().isEmpty()) {
+      return Optional.empty();
+    }
+    List<Geolocation> locations = response.results().get(0).locations();
+    if (locations.isEmpty()) {
+      return Optional.empty();
+    }
+    return Optional.of(locations.get(0).latLng());
```

Comment:
With streams and `findFirst()`, I think this could be shortened to a single expression:

```java
return response
    .result()
    .get(0)
    .locations()
    
    // If empty, does nothing.
    // This might also be replaceable with a method reference, but idr. xD
    .map(location -> location.latLng()) 

    // Returns Optional.of() if non-empty.
    //     If empty, returns Optional.empty().
    .findFirst();
                                        
```

In fact, it might be possible to fold the code above this line into this single expression too...

Reply:
Is it safe to `get(0)` without checking for size?

Reply:
The `response.results().get(0)` is safe because of the check you did above this section of code.

That said, I think you can probably do another case of `.map()` to skip the one above too, since `.map()` does nothing if nothing exists and does transformations if there are things.

Reply:
Ah whoops I misunderstood your comment, got it.

Reply:
I tried removing the first check but it gets too convoluted with trying to `findFirst` the result and dealing with the `Optional`.